### PR TITLE
Add local Artifactory setup

### PR DESCRIPTION
**Pull request by RobiNino**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+func createLicenseFile(jfrogHome string) (err error) {
+	log.Println("Creating license...")
+
+	license := os.Getenv(licenseEnv)
+	if license == "" {
+		log.Println("No license provided. Skipping.")
+	}
+	defer func() {
+		e := os.Setenv(licenseEnv, "")
```
Comment (yahavi) [inline main.go:265]:
```suggestion
		e := os.Unsetenv(licenseEnv)
```

File: main.go [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+		if err == nil {
+			err = e
+		} else {
+			log.Println("error when unsetting env: " + e.Error())
```
Comment (yahavi) [inline main.go:269]: e may be nil.
I'm not sure what you're trying to do here.

File: main.go [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+	binDir := filepath.Join(jfrogHome, "artifactory", "app", "bin")
+
+	var cmd *exec.Cmd
+	if isWindows() {
+		cmd = exec.Command(filepath.Join(binDir, "InstallService.bat"))
+	} else {
+		cmd = exec.Command(filepath.Join(binDir, "artifactoryctl"), "start")
+	}
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stderr
```
Comment (yahavi) [inline main.go:137]: Isn't it the default?

File: main.go [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+
+	err = waitForArtifactorySuccessfulPing()
+	if err != nil {
+		return err
+	}
+
+	return setCustomUrlBase()
+}
+
+func renameArtifactoryDir(jfrogHome string) error {
```
Comment (yahavi) [inline main.go:91]: Please add documentation about this function, since it is not clear enough

**Review by yahavi — CHANGES_REQUESTED** [review]
Big like @RobiNino!

* Please add a description in the README file.
* Optional feature request after the merge, if you want to harden the security 💪🏼 - Do all of this in a new random user with a random password. Switch back to the original user after everything is done.

File: main.go (hunk 3 above)
Comment (RobiNino) [reply main.go:137]: No:
```
	// If either is nil, Run connects the corresponding file descriptor
	// to the null device (os.DevNull).

File: main.go (hunk 2 above)
Comment (RobiNino) [reply main.go:269]: Wanted to log the error if both e and err are not nil. Added a nil check.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
* I think it'll be better if the gradle and the go code will be located in separate directories under the root fo the project. This way, you'll have the go.mod and go.sum on the root of a sub project, and the same goes for the build.gradle and settins.gradle.
* The README of the project should be up to date with the new content of this project.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Comment by RobiNino** [comment]
Done, you are welcome to review again @eyalbe4 , @yahavi .
I named the subdirectory `local-rt-setup` so we can install it using `go install`. `go install` names the binary by the last directory name (doesn't support providing a different name). Calling the directory `go` would cause confusion with the actual `go` binary.

**Review by yahavi — APPROVED** [review]
LGTM

**Pull request merged**